src/sentry/static/sentry/app/actionCreators/performance.tsx

```diff
@@ -0,0 +1,71 @@
+import {Client} from 'app/api';
+import {t} from 'app/locale';
+import {addErrorMessage} from 'app/actionCreators/indicator';
+
+export function saveKeyTransaction(
+  api: Client,
+  orgId: string,
+  projects: number[],
+  transactionName: string
+): Promise<undefined> {
+  const promise: Promise<undefined> = api.requestPromise(
+    `/organizations/${orgId}/key-transactions/`,
+    {
+      method: 'POST',
+      query: {
+        project: projects.map(id => String(id)),
+      },
+      data: {transaction: transactionName},
+    }
+  );
+
+  promise.catch(response => {
+    if (
+      response &&
+      response.responseJSON &&
+      response.responseJSON.non_field_errors &&
+      response.responseJSON.non_field_errors.length &&
+      response.responseJSON.non_field_errors[0]
+    ) {
+      addErrorMessage(response.responseJSON.non_field_errors[0]);
+    } else {
+      addErrorMessage(t('Unable to update key transaction'));
+    }
+  });
+
+  return promise;
+}
+
+export function deleteKeyTransaction(
+  api: Client,
+  orgId: string,
+  projects: number[],
+  transactionName: string
+): Promise<undefined> {
+  const promise: Promise<undefined> = api.requestPromise(
+    `/organizations/${orgId}/key-transactions/`,
+    {
+      method: 'DELETE',
+      query: {
+        project: projects.map(id => String(id)),
+      },
+      data: {transaction: transactionName},
+    }
+  );
+
+  promise.catch(response => {
+    if (
+      response &&
+      response.responseJSON &&
+      response.responseJSON.non_field_errors &&
+      response.responseJSON.non_field_errors.length &&
+      response.responseJSON.non_field_errors[0]
```

Comment:
Seems like a place you could use `.?` to shorten the code.

Reply:
Done in https://github.com/getsentry/sentry/pull/17954/commits/745943328d30ba09db50de2ad8ca76bb577ca7dc